robot/impl/local_robot_test.go

```diff
@@ -4487,7 +4489,7 @@ func TestRemovingOfflineRemotes(t *testing.T) {
 	// Ensure that the remote is not marked for removal while trying to connect to the remote
 	remote, ok := localRobot.manager.resources.Node(remoteName)
 	test.That(t, ok, test.ShouldBeTrue)
-	test.That(t, remote.MarkedForRemoval(), test.ShouldBeTrue)
+	test.That(t, remote.MarkedForRemoval(), test.ShouldBeFalse)
```

Comment:
not sure why this was true to begin with, the comment says this should be false. In any case, false makes more sense - we shouldn't have gotten into Reconfigure yet at this point

Reply:
Pretty scary this was passing!

Reply:
Agree; how could it have been passing?

Reply:
I think pre-PR, the Reconfigure function can progress until after the remote is marked for removal before it needs the lock, which means it can be true